publisher.go

```diff
@@ -14,17 +14,15 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package kubewatcher
+package fission
 
-import (
-	"k8s.io/client-go/1.5/pkg/watch"
-)
+import "net/http"
 
 type (
 	Publisher interface {
-		// Publish an event to a "target".  Target's meaning depends on the
+		// Publish an request to a "target".  Target's meaning depends on the
 		// publisher: it's a URL in the case of a webhook publisher, or a queue
 		// name in a queue-based publisher such as NATS.
-		Publish(event watch.Event, target string)
+		Publish(req *http.Request)
```

Comment:
The intent of this interface is to abstract over HTTP calls as well as async messages via a message queue.  So we don't want it to have an http request in the interface.  

How about: `Publish(body string, headers map[string]string, target string)` (we'd still need to serialize that into a byte stream later, but we can discuss that in another change).